gatsby/gatsby-plugin-sitemap/gatsby-plugin-sitemap.test.js

```diff
@@ -0,0 +1,116 @@
+const {
+  resolvePages,
+  serialize,
+  filterPages,
+} = require('./gatsby-plugin-sitemap');
+
+const EXAMPLES_NODES = {
+  allSitePage: {
+    nodes: [
+      { path: '/some-page/' },
+      { path: '/de/some-page/' },
+      { path: '/de/only-german/' },
+      { path: '/only-default/' },
+      ,
+      { path: '/ignore-me/' },
+    ],
+  },
+};
+
+const createPagesNodes = (nodes) => {
+  return { allSitePage: { nodes } };
+};
+describe('Gatsby Plugin Sitemap', () => {
+  test('works', () => {
+    const allPages = resolvePages(EXAMPLES_NODES);
+    const excludes = ['/ignore-me/'];
+    const filteredPages = allPages.filter(
+      (page) =>
+        !excludes.some((excludedRoute) => filterPages(page, excludedRoute)),
+    );
+    const result = filteredPages.map((page) => serialize(page));
+    expect(result).toMatchSnapshot();
```

Comment:
One of a few situations where a snapshot makes sense. I usually don't like it because it hides the actual "testing idea" behind a blob of data that is compared.